extensions-core/s3-extensions/src/main/java/io/druid/storage/s3/S3StorageDruidModule.java

```diff
@@ -102,15 +109,44 @@ public void configure(Binder binder)
 
   @Provides
   @LazySingleton
-  public RestS3Service getRestS3Service(AWSCredentialsProvider provider)
+  public AmazonS3 getAmazonS3Client(
+      AWSCredentialsProvider provider,
+      AWSProxyConfig proxyConfig,
+      AWSEndpointConfig endpointConfig
+  )
   {
-    if (provider.getCredentials() instanceof AWSSessionCredentials) {
-      return new RestS3Service(new AWSSessionCredentialsAdapter(provider));
-    } else {
-      return new RestS3Service(new AWSCredentials(
-          provider.getCredentials().getAWSAccessKeyId(),
-          provider.getCredentials().getAWSSecretKey()
-      ));
+    // AmazonS3ClientBuilder can't be used because it makes integration tests failed
+    final ClientConfiguration configuration = new ClientConfigurationFactory().getConfig();
+    final AmazonS3Client client = new AmazonS3Client(provider, setProxyConfig(configuration, proxyConfig));
+
+    if (StringUtils.isNotEmpty(endpointConfig.getUrl())) {
+      if (StringUtils.isNotEmpty(endpointConfig.getServiceName()) &&
+          StringUtils.isNotEmpty(endpointConfig.getSigningRegion())) {
+        client.setEndpoint(endpointConfig.getUrl(), endpointConfig.getServiceName(), endpointConfig.getSigningRegion());
+      } else {
+        client.setEndpoint(endpointConfig.getUrl());
+      }
+    }
+
+    client.setS3ClientOptions(S3ClientOptions.builder().enableForceGlobalBucketAccess().build());
+
+    return client;
+  }
+
+  private static ClientConfiguration setProxyConfig(ClientConfiguration conf, AWSProxyConfig proxyConfig)
+  {
+    if (StringUtils.isNotEmpty(proxyConfig.getHost())) {
+      conf.setProxyHost(proxyConfig.getHost());
+    }
+    if (proxyConfig.getPort() != -1) {
```

Comment:
`-1` means "use default" in java URL world if I recall. does this need to be an invalid proxy number like `-2`?

Reply:
AWS's default proxy port is `-1`, so this means we will use the default configuration of AWS. I added a comment about this.